Added Heltun HE-FT01 Wall Heating Thermostat to Database

BorisFierlings opened the pull request:
I've added the heft01_0_0.xml definition file for the Heltun HE-FT01 Wall Heating Thermostat to Database

cdjackson commented:
Please an you add this to the database -:

https://opensmarthouse.org/zwavedatabase/

If I merge this it will simply be overwritten when the next database export is done.

cdjackson commented:
I'm not sure what the error was, but it does look like it generated the outline database entry -:

https://opensmarthouse.org/zwavedatabase/1658

BorisFierlings commented:
Thanks, found the entry and have startet to complete it.

BorisFierlings commented:
Hi Chris,

how can I add command classes to the endpoint. There are two missing.
COMMAND_CLASS_THERMOSTAT_FAN_STATE V2
COMMAND_CLASS_CLOCK V1

BorisFierlings commented:
Hi Chris,
I've still problems to create this device in the zwavedatabase. I'v delete the corrupt endpoint and tried to create a new one. It says successful saved, but there is still no endpoint. How to proceed?

apella12 commented:
What does the OH generated XML in the userdata/zwave folder that you are uploading into the ZW DB look like?

I have seen the first problem you had with missing CCs occur before.  I don't recall the details but believe there was a CC that wasn't in the acceptable list and it errors, but the shows all the CC's before the problem, but doesn't include the ones afterward, so not a complete upload.  My thought is to look at your XML, hopefully find the CC in question and then reload as a new device without it.  You can copy the parameters and Association groups from the messed up one to the new one.

BorisFierlings commented:
Hi Robert, 
thanks for your support!
I've renamed the attached xml to txt, so that I can attach it.
[network_fe03e9cd__node_205.txt](https://github.com/user-attachments/files/20515990/network_fe03e9cd__node_205.txt)
I' also attached the thing definition, I use in my openhab installation  in a patched zwave archive. Also renamed from xml to txt.
[heft01_0_0.txt](https://github.com/user-attachments/files/20516018/heft01_0_0.txt)

apella12 commented:
Ok. It was the Fan Modes. Here is what I [did.](https://www.opensmarthouse.org/zwavedatabase/1664)  Please review and mark for review on the overview tab, once you are satisfied.

What needs to happen to get all the fan modes is a binding PR on the channels XML.  I'm not sure exactly if the ZW DB needs a similar modification or the device check loads the current binding channels.xml.  Anyway Chris will know.
```
    <!-- Thermostat Fan Mode Channel -->
    <channel-type id="thermostat_fan_mode">
        <item-type>Number</item-type>
        <label>Fan Mode</label>
        <description>Sets the fan mode</description>
        <category>Temperature</category>
        <tags>
            <tag>Control</tag>
            <tag>Airflow</tag>
        </tags>
        <state>
            <options>
                <option value="0">Auto</option>
                <option value="1">On</option>
                <option value="2">Auto High</option>
                <option value="3">On High</option>
                <option value="4">Unknown(4)</option>  // Auto Medium
                <option value="5">Unknown(5)</option>  // Medium
                <option value="6">Circulate</option>
            </options>
        </state>
    </channel-type>
```

apella12 commented:
Learned something new when looking at your changes (via the OH2 XML option). The ZW DB creates a custom subchannel based on the options for Fan Mode and Fan state, so no PR will be required to change the channel.xml (I don't think).  Odd that it didn't do that with the OH XML upload in the first place.

BorisFierlings commented:
Hi, I'm trying to edit your work, but working with this database is awful! It's not working and it's crashing all the time. And after a crash it take hours until the site is available again! Is it the same for you?

apella12 commented:
> Is it the same for you?

No everything seems fine (knock on wood).  

I did look at it yesterday.  Only item I would correct is to delete the extra auto medium under fan mode. Also based on the observation above, the config entries for fan_mode and fan_state appear to be ignored, so could be deleted for neatness (or not). You could verify by testing the DB extracted XML and overwriting the one in your jar. Lastly, I would change the category on the overview to thermostat, but that has no effect. Don't forget to mark for review when you are done (You can still make changes afterward)

BorisFierlings commented:
And again, I only clicked on delete at the extra auto medium - and it crashed. Site is not available now.

BorisFierlings commented:
For me [https://www.opensmarthouse.org](https://www.opensmarthouse.org/)/ is still offline

apella12 commented:
I'm sorry, but I made it worse.  It deleted the channel with the idea to recreate it, but it seems to be stuck.  I'll try again in a bit. Last resort is to start over completely, again :-(
